### PR TITLE
LPS-89710 Folder CRUD implementation

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+			).param(
+				"description", ""
+			).param(
+				"type", 1
+			).param(
+				"manualMembership", true
+			).param(
+				"membershipRestriction", 0
+			).param(
+				"friendlyURL", "/" + groupName
```
Comment (cgoncas) [inline modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java]: maybe we could use random values

File: modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+		).auth(
+		).preemptive(
+		).basic(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/json"
+		).header(
+			"Content-Type", "application/json"
+		).body(
+			"{\"description\":\"testDeleteFolder description\"," +
```
Comment (cgoncas) [inline modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java]: Use DTO, with jackson object mapper

File: modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+			new URL(_url, "/api/jsonws/group/delete-group")
+		).then(
+		).statusCode(
+			200
+		);
+	}
+
+	@Test
+	public void testDeleteFolder() throws MalformedURLException {
+		int folderId = RestAssured.given(
```
Comment (cgoncas) [inline modules/apps/headless/headless-document-library/headless-document-library-client-test/src/testIntegration/java/com/liferay/headless/document/library/client/test/FolderResourceTest.java]: extract to common method

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**